plugins/kotlin/j2k/shared/src/org/jetbrains/kotlin/nj2k/expressions.kt

```diff
@@ -142,6 +142,10 @@ fun JKAnnotationList.annotationByFqName(fqName: String): JKAnnotation? =
 
 fun stringLiteral(content: String, typeFactory: JKTypeFactory): JKExpression {
     val lines = content.split('\n')
+    if (lines.size > 1 && lines.last().isNotEmpty()) {
```

Comment:
I don't understand the significance of the condition `&& lines.last().isNotEmpty()`. When I comment it out, no tests fail. Is it necessary? Also, can this function be simplified now? It looks like we care about two cases: when there is a single line, and when there are many lines. If some cases are missing from the test data, please add them.

Note: make sure to run K2 J2K tests and remove the "IGNORE_K2" directive where it is not necessary now, as described in the meetings document.

Reply:
Originally I was writing this because in the current test cases, we don't want to convert:
`val x = "hello" + "\n"`
to a raw string. However, on further thought I think it might make sense to convert to a raw string for `@deprecated`  doc tags regardless. I will simplify this section.